src/app/models/work-item.ts

```diff
@@ -104,4 +113,81 @@ export interface WorkItemUI {
   createdAt: string;
   labels: LabelUI[];
   comments: CommentUI[];
+  wiState: string;
+  wiDescription: string;
+}
+
+export class WorkItemMapper implements Mapper<WorkItemService, WorkItemUI> {
+  itMapper = new IterationMapper();
+  wiTypeMapper = new WorkItemTypeMapper();
+  areaMapper = new AreaMapper();
+  userMapper = new UserMapper();
+  lMapper = new LabelMapper();
+  cMapper = new CommentMapper();
+  
+  serviceToUiMapTree: MapTree = [{
+      fromPath: ['id'],
+      toPath: ['id']
+    }, {
+      fromPath: ['attributes','system.title'],
+      toPath: ['title']
+    }, {
+      fromPath: ['attributes','system.number'],
+      toPath: ['sysNumber']
+    }, {
+      fromPath: ['attributes','createdAt'],
+      toPath: ['createdAt']
+    }, 
+  ];
+
+  uiToServiceMapTree: MapTree = [{
+      toPath: ['id'],
+      fromPath: ['id']
+    }, {
+      toPath: ['attributes','system.title'],
+      fromPath: ['title']
+    }, {
+      toPath: ['type'],
+      toValue: 'workitems'
+    }, {
+      toPath: ['attributes','system.state'],
+      fromPath: ['wiState']
+    }
+  ];
+  
+  toUIModel(arg: WorkItemService): WorkItemUI {
+    let workItemUI = switchModel<WorkItemService, WorkItemUI>(
+      arg, this.serviceToUiMapTree
+    );
+    return this.tempToUIModel(arg, workItemUI);
+  }
+
+  tempToUIModel(arg: WorkItemService, workItemUI: WorkItemUI): WorkItemUI {
+    workItemUI.area = this.areaMapper.toUIModel(arg.relationalData.area);
+    workItemUI.assignees = arg.relationalData.assignees.map(a => this.userMapper.toUIModel(a));
+    workItemUI.creator = this.userMapper.toUIModel(arg.relationalData.creator);
+    workItemUI.comments = arg.relationalData.comments.map(c => this.cMapper.toUIModel(c));
+    workItemUI.iteration = this.itMapper.toUIModel(arg.relationalData.iteration);
+    workItemUI.labels = arg.relationalData.labels.map(l => this.lMapper.toUIModel(l));
+    workItemUI.type = this.wiTypeMapper.toUIModel(arg.relationalData.wiType);
+    return workItemUI;
+  } 
```

Comment:
This has been done so that derived types can be mapped individually and then return the object that will have all the mapped properties